[GStreamer][MediaStream] Move mock display source to its own cpp file

philn opened the pull request:
#### db5d990d229c042c716d4b69a80eb28d6f5dfb3a
<pre>
[GStreamer][MediaStream] Move mock display source to its own cpp file
<a href="https://bugs.webkit.org/show_bug.cgi?id=258072">https://bugs.webkit.org/show_bug.cgi?id=258072</a>

Reviewed by Xabier Rodriguez-Calvar.

Moving code around for consistency purposes.

* Source/WebCore/platform/SourcesGStreamer.txt:
* Source/WebCore/platform/mediastream/gstreamer/MockDisplayCaptureSourceGStreamer.cpp: Copied from Source/WebCore/platform/mediastream/gstreamer/MockRealtimeVideoSourceGStreamer.cpp.
(WebCore::MockDisplayCaptureSourceGStreamer::create):
(WebCore::MockDisplayCaptureSourceGStreamer::MockDisplayCaptureSourceGStreamer):
(WebCore::MockDisplayCaptureSourceGStreamer::~MockDisplayCaptureSourceGStreamer):
(WebCore::MockDisplayCaptureSourceGStreamer::stopProducingData):
(WebCore::MockDisplayCaptureSourceGStreamer::requestToEnd):
(WebCore::MockDisplayCaptureSourceGStreamer::setMuted):
(WebCore::MockDisplayCaptureSourceGStreamer::videoFrameAvailable):
(WebCore::MockDisplayCaptureSourceGStreamer::capabilities):
(WebCore::MockDisplayCaptureSourceGStreamer::settings):
* Source/WebCore/platform/mediastream/gstreamer/MockDisplayCaptureSourceGStreamer.h: Copied from Source/WebCore/platform/mediastream/gstreamer/MockRealtimeVideoSourceGStreamer.h.
* Source/WebCore/platform/mediastream/gstreamer/MockRealtimeVideoSourceGStreamer.cpp:
(WebCore::MockDisplayCaptureSourceGStreamer::create): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::MockDisplayCaptureSourceGStreamer): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::~MockDisplayCaptureSourceGStreamer): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::stopProducingData): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::requestToEnd): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::setMuted): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::videoFrameAvailable): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::capabilities): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::settings): Deleted.
* Source/WebCore/platform/mediastream/gstreamer/MockRealtimeVideoSourceGStreamer.h:
* Source/WebCore/platform/mock/MockRealtimeMediaSourceCenter.cpp:

Canonical link: <a href="https://commits.webkit.org/265316@main">https://commits.webkit.org/265316@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9f3ad6e6c938389e039742257637a0591ec70504

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10553 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/10776 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11053 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/12199 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/10125 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/10567 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/13146 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/10759 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13059 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/10714 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/11642 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/8874 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12600 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/8936 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9530 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/16792 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10007 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/9682 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/12935 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/10144 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/8243 "1 flakes 5 failures") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9299 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2532 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13562 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/10001 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

calvaris commented on Source/WebCore/platform/mediastream/gstreamer/MockDisplayCaptureSourceGStreamer.cpp:
```diff
@@ -0,0 +1,128 @@
+
+    if (constraints) {
+        if (auto error = mockSource->applyConstraints(*constraints))
+            return WTFMove(error->message);
```
Moving when returning is useless, you are not going to move anything.

philn replied:
```
In file included from /var/home/phil/WebKit/WebKitBuild/GTK/Release/WebCore/DerivedSources/unified-sources/UnifiedSource-3c72abbe-63.cpp:2:                                                                          
/var/home/phil/WebKit/Source/WebCore/platform/mediastream/gstreamer/MockDisplayCaptureSourceGStreamer.cpp:37:20: error: no viable conversion from returned value of type 'String' to function return type 'CaptureSou
rceOrError'                                                                                                                                                                                                          
            return error->message;                                                                                                                                                                                   
                   ^~~~~~~~~~~~~~                                                                                                                                                                                    
/var/home/phil/WebKit/Source/WebCore/platform/mediastream/RealtimeMediaSource.h:357:8: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'String' to 'const CaptureSou
rceOrError &' for 1st argument                                                                                                                                                                                       
struct CaptureSourceOrError {                                                                                                                                                                                        
       ^                                                                                                                                                                                                             
/var/home/phil/WebKit/Source/WebCore/platform/mediastream/RealtimeMediaSource.h:357:8: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'String' to 'CaptureSourceOrE
rror &&' for 1st argument                                                                                                                                                                                            
struct CaptureSourceOrError {                                                                                                                                                                                        
       ^                                                                                                                                                                                                             
/var/home/phil/WebKit/Source/WebCore/platform/mediastream/RealtimeMediaSource.h:359:5: note: candidate constructor not viable: no known conversion from 'String' to 'Ref<RealtimeMediaSource> &&' for 1st argument   
    CaptureSourceOrError(Ref<RealtimeMediaSource>&& source) : captureSource(std::move<WTF::CheckMoveParameter>(source)) { }                                                                                          
    ^                                                                                                                                                                                                                
/var/home/phil/WebKit/Source/WebCore/platform/mediastream/RealtimeMediaSource.h:360:5: note: candidate constructor not viable: expects an rvalue for 1st argument                                                    
    CaptureSourceOrError(String&& message) : errorMessage(std::move<WTF::CheckMoveParameter>(message)) { }                                                                                                           
    ^                                                                                                                                                                                                                
/var/home/phil/WebKit/WebKitBuild/GTK/Release/WTF/Headers/wtf/text/WTFString.h:237:5: note: candidate function                                                                                                       
    operator UnspecifiedBoolTypeA() const;                                                                                                                                                                           
    ^                                                                                                                                                                                                                
/var/home/phil/WebKit/WebKitBuild/GTK/Release/WTF/Headers/wtf/text/WTFString.h:238:5: note: candidate function                                                                                                       
    operator UnspecifiedBoolTypeB() const;                                                                                                                                                                           
    ^                                                                                                                                                                                                                
1 error generated.
```

calvaris replied:
Gosh, so this is implicitly calling the `CaptureSourceOrError(String&& message)` constructor. Then I guess it's ok but I would prefer to do a more explicit `return CaptureSourceOrError(WTFMove(error->message));`

philn committed:
[GStreamer][MediaStream] Move mock display source to its own cpp file
https://bugs.webkit.org/show_bug.cgi?id=258072

Reviewed by Xabier Rodriguez-Calvar.

Moving code around for consistency purposes.

* Source/WebCore/platform/SourcesGStreamer.txt:
* Source/WebCore/platform/mediastream/gstreamer/MockDisplayCaptureSourceGStreamer.cpp: Copied from Source/WebCore/platform/mediastream/gstreamer/MockRealtimeVideoSourceGStreamer.cpp.
(WebCore::MockDisplayCaptureSourceGStreamer::create):
(WebCore::MockDisplayCaptureSourceGStreamer::MockDisplayCaptureSourceGStreamer):
(WebCore::MockDisplayCaptureSourceGStreamer::~MockDisplayCaptureSourceGStreamer):
(WebCore::MockDisplayCaptureSourceGStreamer::stopProducingData):
(WebCore::MockDisplayCaptureSourceGStreamer::requestToEnd):
(WebCore::MockDisplayCaptureSourceGStreamer::setMuted):
(WebCore::MockDisplayCaptureSourceGStreamer::videoFrameAvailable):
(WebCore::MockDisplayCaptureSourceGStreamer::capabilities):
(WebCore::MockDisplayCaptureSourceGStreamer::settings):
* Source/WebCore/platform/mediastream/gstreamer/MockDisplayCaptureSourceGStreamer.h: Copied from Source/WebCore/platform/mediastream/gstreamer/MockRealtimeVideoSourceGStreamer.h.
* Source/WebCore/platform/mediastream/gstreamer/MockRealtimeVideoSourceGStreamer.cpp:
(WebCore::MockDisplayCaptureSourceGStreamer::create): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::MockDisplayCaptureSourceGStreamer): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::~MockDisplayCaptureSourceGStreamer): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::stopProducingData): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::requestToEnd): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::setMuted): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::videoFrameAvailable): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::capabilities): Deleted.
(WebCore::MockDisplayCaptureSourceGStreamer::settings): Deleted.
* Source/WebCore/platform/mediastream/gstreamer/MockRealtimeVideoSourceGStreamer.h:
* Source/WebCore/platform/mock/MockRealtimeMediaSourceCenter.cpp:

Canonical link: https://commits.webkit.org/265316@main
